diff --git a/ext/standard/microtime.c b/ext/standard/microtime.c
index 46e770de..26d2f6ab 100644
--- a/ext/standard/microtime.c
+++ b/ext/standard/microtime.c
@@ -123,7 +123,8 @@ PHP_FUNCTION(getrusage)
 #ifdef PHP_WIN32 /* Windows only implements a limited amount of fields from the rusage struct */
 	PHP_RUSAGE_PARA(ru_majflt);
 	PHP_RUSAGE_PARA(ru_maxrss);
-#elif !defined(_OSD_POSIX) && !defined(__HAIKU__)
+/* GOPHPER: wasi-libc's struct rusage has only ru_utime and ru_stime. */
+#elif !defined(_OSD_POSIX) && !defined(__HAIKU__) && !defined(__wasi__)
 	PHP_RUSAGE_PARA(ru_oublock);
 	PHP_RUSAGE_PARA(ru_inblock);
 	PHP_RUSAGE_PARA(ru_msgsnd);
